### PR TITLE
Update ac tool to support ansible-lint 6.22 and expanded beyond RSA support. 

### PR DESCRIPTION
**Pull request by ddimatos**
Addresses #1325 and also expands beyond RSA support, includes ansible-core 2.17-beta support.

##### SUMMARY
- Support for ansible-lint 6.22 was added to all managed venv's
- Support for additional algorithm support because some of the systems don't agree on RSA as a mutual signature algorithm.
- Added a venv template for ansible-core 2.17-beta
- Added the option to choose which venv to start based on its name (also added for stopping a venv)
- Added support to create a managed venv of the latest devel source in the community for ansible-core.
- fixed a script issues location awareness that resulting in error when run from another script, added a env for 2.17.

For example:` ./ac --venv-start --name venv-latest` results in `ansible-playbook [core 2.18.0.dev0]`

```
~/git/gh/ibm_zos_core $: ./ac --venv-start --name venv-latest
-----------------------------------------------------------------------
Starting managed python virtual environment: venv-latest
-----------------------------------------------------------------------
(venv-latest) ansible-playbook --version
[WARNING]: You are running the development version of Ansible. You should only run Ansible from "devel" if you are modifying the Ansible engine, or trying out features under development. This is a rapidly changing
source of code and can become unstable at any point.
ansible-playbook [core 2.18.0.dev0]
  config file = /Users/ddimatos/git/gh/ibm_zos_core/ansible.cfg
  configured module search path = ['/Users/ddimatos/.ansible/collections/ansible_collections/ibm/ibm_zos_core/plugins/modules']
  ansible python module location = /Users/ddimatos/git/gh/ibm_zos_core/venv/venv-latest/lib/python3.12/site-packages/ansible
  ansible collection location = /Users/ddimatos/.ansible/collections:/usr/share/ansible/collections
  executable location = /Users/ddimatos/git/gh/ibm_zos_core/venv/venv-latest/bin/ansible-playbook
  python version = 3.12.0 (main, Oct  2 2023, 12:03:24) [Clang 15.0.0 (clang-1500.0.40.1)] (/Users/ddimatos/git/gh/ibm_zos_core/venv/venv-latest/bin/python3.12)
  jinja version = 3.1.3
  libyaml = True
```

image

##### ISSUE TYPE
- Feature Pull Request

### CODE REVIEW

File: ac [hunk 1]
```diff
@@ -635,11 +663,24 @@ venv_setup(){
 # TODO: Allow user to specify which venv they can start
 # ------------------------------------------------------------------------------
 #->venv-start:
-## Activate the lastest ansible managed virtual environment.
-## Usage: ac [--venv-start]
+## Activate the lastest ansible managed virtual environment or optionally start
```
Comment (richp405) [inline ac:666]: nit-pick: latest, instead of 'lastest'

File: ac [hunk 2]
```diff
@@ -650,14 +691,27 @@ venv_start(){
 # TODO: Allow user to specify which venv they can stop
 # ------------------------------------------------------------------------------
 #->venv-stop:
-## Deactivate the lastest ansible managed virtual environment.
-## Usage: ac [--venv-stop]
+## Deactivate the lastest ansible managed virtual environment or optionally deactivate
```
Comment (richp405) [inline ac:694]: nit-pick: latest, instead of 'lastest'

File: scripts/mounts.env [hunk 3]
```diff
@@ -28,21 +28,21 @@
...
-"11:1.2.1:/zoau/v1.2.1:IMSTESTU.ZOAU.V121.ZFS "\
-"12:1.2.2:/zoau/v1.2.2:IMSTESTU.ZOAU.V122.ZFS "\
-"13:1.2.3:/zoau/v1.2.3:IMSTESTU.ZOAU.V123.ZFS "\
-"14:1.2.4:/zoau/v1.2.4:IMSTESTU.ZOAU.V124.ZFS "\
-"15:1.2.5:/zoau/v1.2.5:IMSTESTU.ZOAU.V125.ZFS "\
-"16:1.3.0:/zoau/v1.3.0:IMSTESTU.ZOAU.V130.ZFS "\
-"17:latest:/zoau/latest:IMSTESTU.ZOAU.LATEST.ZFS "
+zoau_mount_list_str="1:1.0.0-ga:/zoau/v1.0.0-ga:IMSTESTU.ZOAU.V100.GA.ZFS "\
+"2:1.0.1-ga:/zoau/v1.0.1-ga:IMSTESTU.ZOAU.V101.GA.ZFS "\
+"3:1.0.2-ga:/zoau/v1.0.2-ga:IMSTESTU.ZOAU.V102.GA.ZFS "\
```
Comment (richp405) [inline scripts/mounts.env:33]: ZOAU.V102.GA.ZFS is now used for 1.2.5.8, 'latest' is a misnomer, and relates to 1.2.5.6, but I haven't changed the markings yet

**Review by richp405 — CHANGES_REQUESTED** [review]
Only minor nitpicks on comment typos, but I will request changes on the share list.  We now have 1.2.5.8 available.

File: ac (hunk 1 above)
Comment (ddimatos) [reply ac:666]: Done thank you.

File: ac (hunk 2 above)
Comment (ddimatos) [reply ac:694]: Done Thank you.

File: scripts/mounts.env (hunk 3 above)
Comment (ddimatos) [reply scripts/mounts.env:33]: Thanks @richp405 this line was supposed to have been removed, I had it below as part of 1.2.5, `"13:1.2.5:/zoau/v1.2.5:IMSTESTU.ZOAU.V102.GA.ZFS "\`, I will update this.

Comment (ddimatos) [reply scripts/mounts.env]: Updated, thanks @richp405

**Review by richp405 — APPROVED** [review]

**Review by fernandofloresg — APPROVED** [review]

**Pull request merged**